### PR TITLE
chore: eccrypto alternative

### PR DESCRIPTION
**Pull request by alexandre-abrioux**
## Description of the changes

This is an attempt to fix https://github.com/RequestNetwork/requestNetwork/issues/1227, and if possible, should also resolve https://github.com/RequestNetwork/requestNetwork/issues/1207.
It contains a new set of tests to ensure the transition from `eccrypto` covers all cases.

### eccrypto-js

What I tried but did **_not_** work was to replace `eccrypto` with `eccrypto-js`, like proposed in https://github.com/RequestNetwork/requestNetwork/issues/1207. Unfortunately, the `eccryptoNativeData` dataset could be decrypted but the `eccryptoBrowserData` could not. It looks like the library contains some of the same faults as `eccrypto` (see https://github.com/pedrouid/eccrypto-js/issues/34)

### @toruslabs/eccrypto

What worked was to replace `eccrypto` with [`@toruslabs/eccrypto`](https://www.npmjs.com/package/@toruslabs/eccrypto) which is a fork of the initial library, re-implemented in Typescript, that fixes the initial bugs, and that contains no native C++ binary. However, there was a requirement to move to Node18.

### CODE REVIEW

**Comment by MantisClone** [comment]
It's worth noting that this change **drops support for Node 16**. It makes sense to do this because [Node 16 security support ended on September 11, 2023](https://endoflife.date/nodejs). Thus, I just closed #1178 as not planned.

**Comment by MantisClone** [comment]
It's worth noting that this change **drops support for Node 16**. It makes sense to do this because [Node 16 security support ended on September 11, 2023](https://endoflife.date/nodejs). Thus, I just closed #1178 as not planned.

File: packages/utils/test/crypto/ec-utils.test.ts [hunk 1]
```diff
@@ -159,6 +161,19 @@ describe('Utils/EcUtils', () => {
         ),
       ).rejects.toThrowError('The encrypted data is not well formatted');
     });
+
+    it('should be compatible with previous implementation of eccrypto', async () => {
+      const dataTypes = { native: eccryptoNativeData, browser: eccryptoBrowserData } as const;
+      for (const dataType in dataTypes) {
```
Comment (benjlevesque) [inline packages/utils/test/crypto/ec-utils.test.ts:167]: use it.each?

**Review by benjlevesque — APPROVED** [review]
great work 👏

File: packages/utils/test/crypto/ec-utils.test.ts (hunk 1 above)
Comment (alexandre-abrioux) [reply packages/utils/test/crypto/ec-utils.test.ts]: done in https://github.com/RequestNetwork/requestNetwork/pull/1229/commits/9f792560b62e59850390b1c4011ad1c5a89a87d4

**Review by MantisClone — APPROVED** [review]
Looks good 👍 great work 💯

**Review by olivier7delf — APPROVED** [review]

**Pull request merged**